### ISSUE TITLE
OpenShift v4.6.6 (CRC v1.20) cluster stops after 2 days, status:  Not Reachable (v4.6.6)

### ISSUE THREAD

**Issue by rscher** [code-ready/crc#1830]
### General information
same issue reported in #1786 

  * OS: Linux
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```bash
# Put the output of `crc version`
$ crc version
CodeReady Containers version: 1.20.0+ef3f80d
OpenShift version: 4.6.6 (embedded in executable)
  
## CRC status
```bash
# Put the output of `crc status`
$ crc  status
CRC VM:          Running
OpenShift:       Not Reachable (v4.6.6)
Disk Usage:      14.38GB of 32.72GB (Inside the CRC VM)
Cache Usage:     13.69GB
Cache Directory: /home/crcuser/.crc/cache

## CRC config
```bash
# Put the output of `crc config view`
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.8 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.8"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.8 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.8:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.8
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.8"

### Steps to reproduce

  1.  crc setup
  2.  crc start -c 16  -m 32768 -p ~/pull-secret.txt   --log-level debug
  3.  Run script to add insecure registry 
       https://github.com/code-ready/crc/wiki/Adding-an-insecure-registry
  4.  after ~1-2 days, cluster unresponsive
  5.  restart (crc stop;crc start)  does not work, OpenShift still  Not Reachable
       Only crc delete, then steps 1-3 again and then redeploy applications
       

### Expected
$ crc  status
CRC VM:          Running
OpenShift:       Running
Disk Usage:      14.38GB of 32.72GB (Inside the CRC VM)
Cache Usage:     13.69GB
Cache Directory: /home/crcuser/.crc/cache


### Actual
$ crc status --log-level debug
DEBU CodeReady Containers version: 1.20.0+ef3f80d
DEBU OpenShift version: 4.6.6 (embedded in executable)
Checking file: /home/crcuser/.crc/machines/crc/.crc-exist
DEBU About to run SSH command:
df -B1 --output=size,used,target /sysroot | tail -1
DEBU SSH command results: err: <nil>, output: 32720793600 14460198912 /sysroot
DEBU About to run SSH command with hidden output
DEBU SSH command failed
DEBU cannot get OpenShift status: ssh command error:
command : oc get co -ojson --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1
output  :
CRC VM:          Running
OpenShift:       Not Reachable (v4.6.6)
Disk Usage:      14.46GB of 32.72GB (Inside the CRC VM)
Cache Usage:     13.69GB
Cache Directory: /home/crcuser/.crc/cache


### Logs

https://gist.github.com/rscher/c975606f0b3d148446bfd9263bc47a28

**Comment by timroster** [comment]
similar to / duplicate of: https://github.com/code-ready/crc/issues/1786

**Comment by rscher** [comment]
It seems that the stability of crc's embedded VM  is dependant upon the hypervisor of the host.
The same OpenShift v4.6.6 (CRC v1.20) cluster in another environment has been stable and running continuously 
for over 60 days. Previous crc versions in same env, max MTBF observed to be over 90 days.

The main difference appears to be hypervisor / virtualization of the host.
The following compares the hypervisor / virtualization differences between CRC v.120 installed in the 2 envs:
1)  stable environment,        KVM host details: VMware / VT-x / GenuineIntel / Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz
2)  unstable environment,   KVM host details: KVM / AMD-V /  AuthenticAMD / AMD EPYC Processor (with IBPB)

Comparison of environments using  **lscpu** and **dmidecode** cmds
For cmd outputs and details see log:
 https://gist.github.com/rscher/fb8674a5d32ddfb9829a6b40352bee7a

**Comment by gbraad** [comment]
> It seems that the stability of crc's embedded VM is dependant upon the hypervisor of the host.

This is almost like an understatement, but yes.. especially when using
HyperKit (macOS) this causes syncing issues when it concerns time
However, this would suggest something else... We have seen issues
before when it concerned AMD Epyx/Ryzen CPUs.


On Mon, Jan 11, 2021 at 6:33 AM Russell Scher <notifications@github.com> wrote:
>
> It seems that the stability of crc's embedded VM is dependant upon the hypervisor of the host.
> The same OpenShift v4.6.6 (CRC v1.20) cluster in another environment has been stable and running continuously
> for over 60 days. Previous crc versions in same env, max MTBF observed to be over 90 days.
>
> The main difference appears to be hypervisor / virtualization of the host.
> The following compares the hypervisor / virtualization differences between CRC v.120 installed in the 2 envs:
>
> stable environment, KVM host details: VMware / VT-x / GenuineIntel / Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz
> unstable environment, KVM host details: KVM / AMD-V / AuthenticAMD / AMD EPYC Processor (with IBPB)
>
> Comparison of environments using lscpu and dmidecode cmds
> For cmd outputs and details see log:
> https://gist.github.com/rscher/fb8674a5d32ddfb9829a6b40352bee7a
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.



-- 

   Gerard Braad | http://gbraad.nl

   STEM is the source, but BUILD is how you develop!
   [ Better Understanding Involves Learning and Doing ]

**Comment by gbraad** [comment]
https://github.com/code-ready/crc/issues/1035
https://github.com/code-ready/crc/issues/1201
https://github.com/code-ready/crc/issues/527

Though these don't look similar as it was either around LUKS encryption, or nested virtualization.

**Comment by timroster** [comment]
I have observed this problem with KVM hypervisor on both AMD Ryzen (native - e.g. KVM on native OS) and Intel (nested - KVM as virtualization layer for VM, in turn running KVM/libvirt hosting crc). When this problem presents, `etcd` has fallen and it can't get back up... On my currently failing CRC instance (Intel), kubelet has been running for 7 days (Jan 3 start) as has the VM itself.

**Comment by gbraad** [comment]
From the logs we can conclude that both cases rely on nested virtualization, but the unstable case is using an old(er) version of Ubuntu. Please make sure to install recent versions. Also, we only test this on RHEL/CentOS/Fedora for a possible nested setup. Do note, this is not supported by us.

Besides this, this is identical to a previous case from this environment. Let's continue this by email.

**Comment by gbraad** [comment]
@timroster you work with @rscher ?

**Comment by timroster** [comment]
@gbraad same company ;) different organization

**Comment by gbraad** [comment]
> @gbraad same company ;) different organization

File a separate issue as they are likely a little different.

**Comment by timroster** [comment]
I commented and recently updated #1786 . It matches what I've seen.

**Comment by stephmen** [comment]
Hi I am facing the same type of issues on Mac Version 11.1. It use to work so well on version 4.2. I never experimented any issue. I keep rebuilding a large environment and it is getting very discouraging. Is there any fix or solution to come ? 

I need my demo to he showed to our customer but for few weeks now I just keep rebuilding my project from scratch.

**Comment by kitty-catt** [comment]
I have the same issue on:
- qemu-kvm 
- Intel(R) Core(TM) i7-4810MQ 
- VT-x 
- Red Hat Enterprise Linux Workstation 7.9
..

**Comment by kitty-catt** [comment]
I notice that when I rebuild the crc, it says that renewing the Kubelet client certificate can take up to 8 minutes, ... the CRC start on a cluster with the issue exits much earlier.

**Comment by gbraad** [comment]
This issue is about

> KVM host details: KVM / AMD-V / AuthenticAMD / AMD EPYC Processor (with IBPB)

being unstable. By email we confirmed this is related to running on an Ubuntu LTS host that has been confirmed before to be an issue for the nested virtualization. Hence this issue was closed.

Anyone besides the OP please file a new issue providing all the details about the OS this runs on, etc. as requested in the template. This issue was closed and not actively tracked anymore.
